f8a_worker/storages/s3_manual_tagging.py

```diff
@@ -0,0 +1,25 @@
+from . import AmazonS3
+
+
+class S3ManualTagging(AmazonS3):
+    def store_user_data(self, input_json):
+        if input_json.get('user') is None or input_json.get('ecosystem') is None:
+            return {'status': 'failure'}
```

Comment:
Raise an exception here.

Reply:
Also to test existence of a key in dictionary use

```python
if 'user' not in input_json or 'ecosystem' not in input_json:
```